### PR TITLE
Added measured benchmarks for `pallet_xcm`

### PR DESCRIPTION
**Pull request by bkontur**
Fix

Fix

### CODE REVIEW

**Review by KiChjang — APPROVED** [review]
Just fixes yeah? No benchmarks in this PR?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2130504

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2132255

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2132549

File: parachains/runtimes/assets/statemine/src/xcm_config.rs [hunk 1]
```diff
@@ -158,7 +161,9 @@ match_types! {
 			frame_system::Call::set_heap_pages { .. } |
 			frame_system::Call::set_code { .. } |
 			frame_system::Call::set_code_without_checks { .. } |
-			frame_system::Call::kill_prefix { .. }
+			frame_system::Call::kill_prefix { .. } |
+			// Used by pallet_xcm_benchmarks for transact
+			frame_system::Call::remark_with_event { .. }
```
[the remark this answers is not in the document]
Comment (KiChjang) [reply parachains/runtimes/assets/statemine/src/xcm_config.rs:166]: Then we should use another extrinsic on that benchmark.

**Review by KiChjang — APPROVED** [review]
If you pull gav-xcm-v3 now, you'll be able to configure your Transact benchmark to have a different runtime call via the `transact_origin_and_runtime_call` associated function in the XCM generic benchmarks configuration.

**Review by KiChjang — CHANGES_REQUESTED** [review]
.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2134368

**Comment by bkontur** [comment]
> If you pull gav-xcm-v3 now, you'll be able to configure your Transact benchmark to have a different runtime call via the `transact_origin_and_runtime_call` associated function in the XCM generic benchmarks configuration.

@KiChjang 
there a still problem with `transact` benchmarks, because `pallet_session::Call::purge_keys` does not fires events, so the assert fails:
```
                // TODO make better assertion? #4426
		let num_events2 = frame_system::Pallet::<T>::events().len();
		assert_eq!(num_events + 1, num_events2);
```

should we fix that TODO or just to change `purge_keys` to other call, which triggers one event?
possible easy fix would be to add `expected_events_count` to the:
```
                fn transact_origin_and_runtime_call(
		) -> Result<(MultiLocation, <Self as crate::generic::Config<I>>::RuntimeCall, u8), BenchmarkError>;
```


error log:

```
Running `target/release/polkadot-parachain benchmark pallet --steps=50 --repeat=20 '--extrinsic=*' --execution=wasm --wasm-execution=compiled --heap-pages=4096 --json-file=/var/lib/gitlab-runner/builds/zyw4fam_/0/parity/mirrors/cumulus/.git/.artifacts/bench.json '--pallet=pallet_xcm_benchmarks::generic' --chain=statemine-dev --header=./file_header.txt --output=./parachains/runtimes/assets/statemine/src/weights/xcm`
2022-12-08 14:23:27 assembling new collators for new session 0 at #0    
2022-12-08 14:23:27 assembling new collators for new session 1 at #0    
2022-12-08 14:23:28 Starting benchmark: pallet_xcm_benchmarks::generic::report_holding    
2022-12-08 14:23:28 Starting benchmark: pallet_xcm_benchmarks::generic::buy_execution    
2022-12-08 14:23:28 Starting benchmark: pallet_xcm_benchmarks::generic::query_response    
2022-12-08 14:23:28 Starting benchmark: pallet_xcm_benchmarks::generic::transact    
2022-12-08 14:23:28 transact_origin_and_runtime_call    
2022-12-08 14:23:28 panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`', /home/bparity/.cargo/git/checkouts/polkadot-9da13ac14bbabc7b/07de17c/xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs:110:9    
```

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2134978

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2136706

File: parachains/runtimes/assets/statemine/src/weights/xcm/mod.rs [hunk 2]
```diff
@@ -234,6 +234,6 @@ impl<Call> XcmWeightInfo<Call> for StatemineXcmWeight<Call> {
 		Weight::MAX
 	}
 	fn unpaid_execution(_: &WeightLimit, _: &Option<MultiLocation>) -> Weight {
-		XcmGeneric::<Runtime>::unpaid_execution()
+		Weight::from_ref_time(3_111_000 as u64)
```
Comment (KiChjang) [inline parachains/runtimes/assets/statemine/src/weights/xcm/mod.rs:237]: Why is this done here?

Comment (bkontur) [reply parachains/runtimes/assets/statemine/src/weights/xcm/mod.rs:237]: obsolete, like we discussed, regenerating new pallet_xcm_benchmarks::generic with real benchmark for unpaid_execution

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2138203

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2138283

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2138527

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2138589

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2138828

**Comment by bkontur** [comment]
@KiChjang 
`pallet_xcm` and `pallet_xcm_benchmarks::generic` are (re)generated for **statemine/statemint/westmint**
+ with some compile fixes, I guess, this could be merged.

but we need to add fixes for this:
```
/cmd queue -c bench-bot $ pallet collectives-polkadot collectives pallet_xcm
/cmd queue -c bench-bot $ pallet bridge-hub-kusama bridge-hubs pallet_xcm
/cmd queue -c bench-bot $ pallet bridge-hub-rococo bridge-hubs pallet_xcm
```

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2144235

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2144337

File: parachains/runtimes/assets/westmint/src/weights/pallet_xcm.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	}
+	// Storage: ParachainInfo ParachainId (r:1 w:0)
+	fn reserve_transfer_assets() -> Weight {
+		// Minimum execution time: 24_850 nanoseconds.
+		Weight::from_ref_time(25_427_000)
+			.saturating_add(T::DbWeight::get().reads(1))
+	}
+	fn execute() -> Weight {
+		// Minimum execution time: 14_909 nanoseconds.
+		Weight::from_ref_time(15_229_000)
```
Comment (KiChjang) [inline parachains/runtimes/assets/westmint/src/weights/pallet_xcm.rs:75]: You'll notice that westmint is the only runtime that doesn't have `Weight::MAX` here, that means that all other runtimes have disabled any origin from running `XcmExecuteFilter`, and westmint is the only one that did enable `execute`. Is this true?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2152954

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2153228

**Review by KiChjang — APPROVED** [review]

File: parachains/runtimes/assets/westmint/src/weights/pallet_xcm.rs (hunk 3 above)
Comment (bkontur) [reply parachains/runtimes/assets/westmint/src/weights/pallet_xcm.rs:75]: yes, it is intentional, I checked with CGP team

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2153833

**Pull request merged**